### ISSUE TITLE
Installed Release-mode UWP app containing Essentials crashes on launch - need an Init (or further info)

### ISSUE THREAD

**Issue by donaldnp** [xamarin/Essentials#815]
### Description
As per https://forums.xamarin.com/discussion/57179/uwp-immediate-crash-in-release-mode a UWP app which contains Essentials - in this case I'm using Clipboard - crashes on launch. Have included assemblies Xamarin.Essentials.Clipboard, Windows.ApplicationModel.DataTransfer.Clipboard, and Windows.ApplicationModel.DataTransfer.DataPackage, to no avail. Can't see what else would need to be added to resolve this.

### Steps to Reproduce
1. create a minimal UWP app in Release mode, package it and install - should work
2. add Essentials to it and use Clipboard for something
3. add/modify App.Xaml.cs - 
using System.Reflection;
List<Assembly> assembliesToInclude=new List<Assembly>();
assembliesToInclude.Add(typeof(Xamarin.Essentials.Clipboard).GetTypeInfo().Assembly);
assembliesToInclude.Add(typeof(Windows.ApplicationModel.DataTransfer.Clipboard).GetTypeInfo().Assembly);
assembliesToInclude.Add(typeof(Windows.ApplicationModel.DataTransfer.DataPackage).GetTypeInfo().Assembly);
Xamarin.Forms.Forms.Init(e,assembliesToInclude);
4. build in Release mode, should still be working
5. create package and install - https://docs.microsoft.com/en-us/windows/uwp/packaging/packaging-uwp-apps. App crashes a few seconds after launch

### Expected Behavior
App launches successfully

### Actual Behavior
Xamarin splash screen comes up, and few seconds later it crashes, with only the following information in the Windows event log...
Faulting application name: TestZXing.UWP.exe, version: 1.0.0.0, time stamp: 0x5d037120
Faulting module name: Windows.UI.Xaml.dll, version: 10.0.17134.799, time stamp: 0x7514da33
Exception code: 0xc000027b
Fault offset: 0x0084f8ec
Faulting process ID: 0x4d24
Faulting application start time: 0x01d522992f2ec186
Faulting application path: C:\Program Files\WindowsApps\5bdedb44-3f4a-4647-83ba-d7a5ed3dc30a_1.0.0.0_x86__b71kk8c7kdz5g\TestZXing.UWP.exe
Faulting module path: C:\Windows\System32\Windows.UI.Xaml.dll
Report ID: 2b5e2e39-0af7-4f9b-b3fb-ecef08f1931c
Faulting package full name: 5bdedb44-3f4a-4647-83ba-d7a5ed3dc30a_1.0.0.0_x86__b71kk8c7kdz5g
Faulting package-relative application ID: App

### Basic Information

- Version with issue: Essentials 1.1.0
- Last known good version:
- IDE: VS2017 Community
- Platform Target Frameworks:
  - UWP:  target: 17134, minimum:16299
- Nuget Packages: Xamarin.Essentials
- Affected Devices: UWP PC

### Screenshots
N/A

### Reproduction Link
https://smartmanapps.visualstudio.com/_git/TestZXing (Barcode is copied to Clipboard)

**Comment by donaldnp** [comment]
Well, I have an update on this. Someone else noticed that your tests have Clipboard running on the main thread, so when I moved it to my main page it worked! (so same symptoms as missing assembly but different cause) It wouldn't work from my viewmodel though (it's original location), not even wrapped in a BeginInvokeOnMainThread(). So I went to update the repo to reproduce this behaviour for you, but now the packaged app won't work at all, not even with all references to Clipboard commented out. The project is now fundamentally broken in some way, and I don't know why (this is even after re-cloning and ONLY commenting out Clipboard references). It seems that adding Clipboard adds something to the project which then breaks it for package deployment. Not sure now how I managed to get it to work before (faced with having to rewrite the whole thing from scratch to get it working again), but if you can reproduce this behaviour (that it needs to be in main thread), then...
1. there needs to be reference to this in the docs (currently no mention of this)
2. preferably update it in a way that this isn't an issue anymore.

**Comment by donaldnp** [comment]
I have an update on this issue (and have updated the repo with this development). Someone else noticed that your tests have Clipboard being run on the main thread - there is no mention of this limitation in the docs! When I moved it to my main thread (from my viewmodel) it worked! However, I first tried wrapping it in a BeginInvokeOnMainThread in my viewmodel, and that DOESN'T work. Not sure if that reveals a flaw in Clipboard or in BeginInvokeOnMainThread (I tried both Device. and MainThread., and didn't work with either). So this was an issue where it showed the same symptoms as another issue (missing assemblies in package) but had a different cause (Clipboard can't be run in a background thread).

So, there are several take-aways from this that need follow-up,,,
1. Clipboard docs needs to specify that it needs to run on main thread (for UWP at least - I haven't tried other platforms)
2. preferably bake something into Clipboard to remove this limitation. e.g. add a BeginInvokeOnMainThread at the root location of this issue
3. please get the relevant VS team to look at this issue of an app which works in Release mode in VS fails when deployed as a packaged app. It takes a long time to package an app, and the information you get back from Windows when it fails is useless in determining the issue (for a long time I was chasing missing assemblies, as that's what has been the cause before with this message, but it was actually a different cause). If a packaged app is going to fail when deployed, then we need to be seeing that failure in VS Release mode, so as to not waste as much time and have more information about what the issue is.

**Comment by jamesmontemagno** [comment]
Thanks for the follow up!

I have added a PR for the docs: https://github.com/MicrosoftDocs/xamarin-docs/pull/1812

I will see what I can do about the packaging and Win10 messages and ask the teams there.

**Comment by donaldnp** [comment]
Excellent. Thanks you @jamesmontemagno ! In regards to BegingInvokeOnMainThread failing to resolve the issue from the viewmodel, I looked at the other issue referenced in your docs PR, and you mentioned that a conversation was needed about that (accessing the Clipboard from another thread). Not sure if you meant with the team, in the issue, in the forum, or something else, but I'll kick off my side here... :-)

   Some of my viewmodel processes (kicked off via button commanding) update a status message property upon completion, which is bound to a label in the UI. The Clipboard is also updated (within the status message PropertyChanged handler) so that the user can easily paste any such status message into some other place. Another context is scanning a barcode, and having the barcode able to be pasted (has frustrated me in the past when I've used a scanner and not been able to paste the result somewhere).
   One solution (currently not working) is accessing Clipboard in the vm, but wrapped in a BeginInvoleOnMainThread.
   Another solution (currently not implemented) would be being able to bind the Clipboard text to something. e.g. being able to bind the Clipboard text to my status message property.
   Hopefully we can look at implementing one or both of those? i.e. I'm saying there are genuine cases for wanting to update Clipboard text from vm (either directly or indirectly).
